src/Microsoft.DotNet.Build.Tasks/PackageFiles/BuildVersion.targets

```diff
@@ -0,0 +1,15 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  
+  <PropertyGroup>
+    <TodayTimeStamp>$([System.DateTime]::Now.ToString(yyyyMMdd))</TodayTimeStamp>
+    <BuildVersionFile Condition="'$(BuildVersionFile)'==''">$(BaseIntermediateOutputPath)/$(TodayTimeStamp)/BuildVersion.props</BuildVersionFile>
```

Comment:
Why does the TodayTimeStamp need to be a directory? Does that make it easier in some way? As best I can tell it might be simpiler to just have it as part of the filename.